packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -100,13 +113,14 @@ export class Tracing implements InstrumentationListener, SnapshotterDelegate, Ha
       return;
     }
 
+    await this._createTracesDirIfNeeded();
```

Comment:
I think we must assign `this._state` synchronously (could be a dummy value I guess?), otherwise we'll need a separate flag `isStarting`.

Reply:
done